sbi/inference/snpe/snpe_b.py

```diff
@@ -1,123 +1,103 @@
 from __future__ import annotations
+from typing import Callable, Optional
 
 import torch
-from torch import Tensor
+from torch import Tensor, nn
+from torch.utils.tensorboard import SummaryWriter
 
 from sbi.inference.snpe.snpe_base import SnpeBase
 
 
 class SnpeB(SnpeBase):
     def __init__(
         self,
-        simulator,
+        simulator: Callable,
         prior,
-        x_o,
-        num_pilot_sims=100,
-        density_estimator="maf",
-        calibration_kernel=None,
-        use_combined_loss=False,
-        z_score_x=True,
-        simulation_batch_size: int = 1,
-        retrain_from_scratch_each_round=False,
-        discard_prior_samples=False,
-        summary_writer=None,
-        device=None,
+        x_o: Tensor,
+        density_estimator: Optional[nn.Module] = None,
```

Comment:
Related to comment in SnpeA, like e.g., here?

Reply:
Let me know if you find the distinction made between type and default value in the previous comment answers your doubts.